client/src/app/components/sidenav/sidenav.component.ts

```diff
@@ -10,24 +12,73 @@ import { AuthService } from 'src/app/services/auth.service';
 })
 export class SideNavComponent implements OnInit {
   // contains the listing of all sidenav menu items
-  options = SideNavListings;
+  options = SystemNavListings;
   // contains the last option chosen, it defaults to the first
   selectedOption: SideNavOption;
-  currentRole;
-  subscription;
 
-  constructor(private authService: AuthService) {}
+  authSubscription;
+  routeSubscription;
+  loggedInUser;
+  loggedInUserRole;
 
-  ngOnInit(): void {
-    this.selectedOption = this.options[0];
+  roles = {
+    SA: 'System Administrator',
+    IM: 'Inventory Manager',
+    SK: 'Stock Keeper'
+  };
+
+  constructor(private router: Router,
+              private authService: AuthService,
+              private tokenService: TokenService) { }
 
-    this.subscription = this.authService.sharedUser
+  ngOnInit(): void {
+    this.authSubscription = this.authService.getOrgMode().subscribe(orgMode => {
+      if (orgMode) {
+        this.options = OrganizationNavListings;
+      } else {
+        this.options = SystemNavListings;
+      }
+    });
+    this.authSubscription = this.authService.sharedUser
       .subscribe((data) => {
-        this.currentRole = data.role;
+        this.loggedInUser = data.username;
+        this.loggedInUserRole = this.roles[data.role];
       });
+    this.setSelected(this.router.url);
+    this.subscribeSelected();
+  }
+
+  exitOrg(): void {
+    this.authService.turnOffOrgMode();
+  }
+
+  logout(): void {
+    this.tokenService.DeleteToken(); // Delete token when user logout
+    this.authService.setLogOut();   // Extra step - sets the sharedUser data to ''
+    this.router.navigate(['login']); // Redirect user to login/register pager
+    // TODO: Check out if we want to delete also the token from the db, in order to regenerate a new one while logging in
```

Comment:
yeah, this is just so that the token doesn't stay lying around while the user isn't using their pc. Plus, if someone visit the url when they have the token, they will be redirected to the dashboard.